### DISCUSSION TITLE
Unclear why I'm getting an error with IN operator

### DISCUSSION THREAD

**Discussion by chase-lewis-akasa** [sqlalchemy/sqlalchemy#8548] [Usage Questions] [answered]
I have a table I want to delete all the rows that correspond to specific ids, but I get the error below. However, I don't understand why there is an error here. If I do a '==' to a single queueid it works so this seems to be an issue with sqlalchemy or I need to pass the data differently here for some reason. The reason I'm getting an error doesn't make much sense to me. 

uuids is an array of UUID items as shown by the error.

Query
```
        delete_statement = delete(models.QueueReviewChance).where(
            models.QueueReviewChance.queueid.in_(uuids)
        )
```

Error
> IN expression list, SELECT construct, or bound parameter object expected, got [UUID(as_uuid='037e1a81-0c67-47c7-9d78-5e86600fdb4a'), UUID(as_uuid='ee632b30-9fb8-4838-a807-0d6ad2171986'), UUID(as_uuid='18121045-c729-475a-a9c8-aebf92578cf5'), UUID(as_uuid='ef2e5bda-bfa9-4c85-8f32-5b3de094cf0c')].

Table
```
class QueueReviewChance(Base):
    __tablename__ = "marsqueuereviewchance"

    # Primary Key
    username = Column(VARCHAR(128), nullable=False, primary_key=True)

    # Foreign Keys
    queueid = Column(
        UUID(as_uuid=True), ForeignKey("marsqueue.id"), nullable=False, primary_key=True
    )

    chance = Column(REAL, nullable=False, server_default=text("1.0"))

    queue = relationship("Queue")

```

**Comment by CaselIT** [comment]
Hi

> uuids is an array of UUID items as shown by the error.

what class are "UUID items" ?

**Comment by zzzeek** [comment, marked as answer]
it looks like you are sending SQLAlchemy ``UUID()`` type objects as data.  you want to use the Python ``uuid.UUID()`` object.

**Reply by chase-lewis-akasa** [reply]
> it looks like you are sending SQLAlchemy `UUID()` type objects as data. you want to use the Python `uuid.UUID()` object.

Yeah in this particular file I had already imported UUID from postgres so got the types mixed up.

**Reply by CaselIT** [reply]
that seemed the case, but I wanted to make sure before suggesting otherwise
